bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/remote/ApiClientImpl.kt

```diff
@@ -0,0 +1,161 @@
+package io.bucketeer.sdk.android.internal.remote
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import io.bucketeer.sdk.android.internal.logd
+import io.bucketeer.sdk.android.internal.model.Event
+import io.bucketeer.sdk.android.internal.model.User
+import io.bucketeer.sdk.android.internal.model.request.GetEvaluationsRequest
+import io.bucketeer.sdk.android.internal.model.request.RegisterEventsRequest
+import io.bucketeer.sdk.android.internal.model.response.ErrorResponse
+import io.bucketeer.sdk.android.internal.model.response.GetEvaluationsResponse
+import io.bucketeer.sdk.android.internal.model.response.RegisterEventsResponse
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.util.concurrent.TimeUnit
+
+internal class ApiClientImpl(
+  endpoint: String,
+  private val apiKey: String,
+  private val featureTag: String,
+  private val moshi: Moshi
+) : ApiClient {
+
+  private val endpoint = endpoint.toHttpUrl()
+
+  private val client: OkHttpClient = OkHttpClient.Builder()
+    .build()
+
+  private val errorResponseJsonAdapter: JsonAdapter<ErrorResponse> by lazy {
+    moshi.adapter(ErrorResponse::class.java)
+  }
+
+  override fun fetchEvaluations(
+    user: User,
+    userEvaluationsId: String,
+    timeoutMillis: Long?
+  ): GetEvaluationsResult {
+    val body = GetEvaluationsRequest(
+      tag = featureTag,
+      user = user,
+      user_evaluations_id = userEvaluationsId
+    )
+
+    val request = Request.Builder()
+      .url(
+        endpoint.newBuilder()
+          .addPathSegments("v1/gateway/evaluations")
+          .build()
+      )
+      .applyHeaders()
+      .post(body = body.toJsonRequestBody())
+      .build()
+
+    val actualClient = if (timeoutMillis == null) {
+      client
+    } else {
+      client.newBuilder()
+        .callTimeout(timeoutMillis, TimeUnit.MILLISECONDS)
+        .build()
+    }
+
+    val result = actualClient.newCall(request).runCatching {
+      logd { "--> Fetch Evaluation\n$body" }
+
+      val (millis, data) = measureTimeMillisWithResult {
+        val rawResponse = execute()
+
+        if (!rawResponse.isSuccessful) {
+          throw rawResponse.toBKTException(errorResponseJsonAdapter)
+        }
+
+        val response = requireNotNull(rawResponse.fromJson<GetEvaluationsResponse>())
+
+        response to (rawResponse.body?.contentLength() ?: -1).toInt()
+      }
+
+      val (response, contentLength) = data
+
+      logd { "--> END Fetch Evaluation" }
+      logd { "<-- Fetch Evaluation\n$response\n<-- END Evaluation response" }
+
+      GetEvaluationsResult.Success(
+        value = response,
+        millis = millis,
+        sizeByte = contentLength,
+        featureTag = featureTag,
+        state = response.data.state
```

Comment:
Unlike v1 implementation, I'm planning to return both response body & latency metrics so that BKTClient can track latency events without using a callback workaround below.

https://github.com/ca-dp/bucketeer-android-sdk/blob/1fbf26e07e979a19529ab448178143e8a46692dc/bucketeer/src/main/java/jp/bucketeer/sdk/BucketeerImpl.kt#L64-L92